File: Day-12-of-30-Days/Solution.java
// Day 12 of 30 Days 👾

public class Solution { // time complexity = O(n^2)
    public static boolean isAnagram(String str1, String str2) {
        String text1 = str1.replaceAll("[^a-zA-Z]", "").toLowerCase();
        String text2 = str2.replaceAll("[^a-zA-Z]", "").toLowerCase();

        if (text1.length() == text2.length()) {
            int count = 0;
            for (int i = 0; i < text1.length(); i++) {
                for (int j = 0; j < text2.length(); j++) {
                    if (text1.charAt(i) == text2.charAt(j)) {
                        count++;
                        break;
                    }
                }
            }
            if (count == text1.length()) {
                return true;
            }
        }
        return false;
    }

    // Examples
    public static void main(String[] args) {
        System.out.println(isAnagram("test", "test")); // true
        System.out.println(isAnagram("Listen", "Silent")); // true
        System.out.println(isAnagram("hello", "world")); // false
        System.out.println(isAnagram("Astronomer", "Moon starer")); // true
    }
}

File: Day-12-of-30-Days/Solution.js
// Day 12 of 30 Days 👾

function isAnagram(str1, str2) {
    let s1 = str1.replace(/[^a-zA-Z]/g, "").toLowerCase();
    let s2 = str2.replace(/[^a-zA-Z]/g, "").toLowerCase();

    if(str1.length == str2.length) {
        let count = 0;
        for(let i=0; i<str1.length; i++) {
            for(let j=0; j<str2.length; j++) {
                if(str1[i] == str2[j]) {
                    count++;
                    break;
                }
            }
        }
        if(count == str1.length) {
            return true;
        }
    }
    return false;
}


// Example s
console.log(isAnagram("hello", "world")); // false
console.log(isAnagram("test", "test")); // true
console.log(isAnagram("boy", "boys")); // false